plugins/nf-lamin/src/main/nextflow/lamin/api/LaminInstance.groovy

```diff
@@ -0,0 +1,98 @@
+package nextflow.lamin.api
+
+import java.util.UUID;
+
+import groovy.transform.CompileStatic
+import groovy.util.logging.Slf4j
+
+import ai.lamin.lamin_api_client.ApiClient;
+import ai.lamin.lamin_api_client.ApiException;
+import ai.lamin.lamin_api_client.Configuration;
+import ai.lamin.lamin_api_client.model.*;
+import ai.lamin.lamin_api_client.api.DefaultApi;
+
+@CompileStatic
+class LaminInstance {
+    final protected LaminHub hub;
+
+    final protected LaminInstanceSettings settings;
+    final protected DefaultApi apiInstance;
+
+    LaminInstance(
+        LaminHub hub,
+        String owner,
+        String name
+    ) {
+        this.hub = hub;
```

Comment:
This should probably be after argument validation?

Reply:
Potato potato, no?